heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java

```diff
@@ -90,6 +95,7 @@ public void call()
    * Handler to activate a topology
    */
   private void activateTopologyHandler(String topologyName) throws TMasterException {
+    assert isRunning;
```

Comment:
Here the topology might not be in the running state?

Reply:
"Running" means there are both topology defn and execution state data.

If any of them is not available, it means the topology is not "running". This could be caused by a failed submit command or a failed kill command.

Like we discussed before, in this state, kill command won't work. User has to learn how heron stores data in different places and checks/clears them manually, which is not good. In this PR, kill command is allowed with warning messages.

Reply:
And for all other commands, validation code shouldn't allow them to be executed. So in this line, isRunning flag should always be true.

Reply:
Karthik mentioned a good point. `Running` can be confusing since we have used it already for topology state `Running`, `Paused`, etc.
Maybe we could rename it to `isExisting`?

Reply:
That is true. Let me think about it. Thanks.